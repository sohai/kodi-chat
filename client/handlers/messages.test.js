import messagesHandler from './messages';

const item = {
  message: 'test',
  username: 'username'
};

describe('(handler) messages', () => {
  it('have right stucture', () => {
    expect(messagesHandler).toMatchSnapshot();
  });

  describe('actions crators corect objects', () => {
    const { actions } = messagesHandler;
    it('actions.addMessage', () => {
      expect(actions.addMessage()).toMatchSnapshot();
    });
    it('actions.incMessage', () => {
      expect(actions.incMessage()).toMatchSnapshot();
    });
    it('actions.removeUserLastMessage', () => {
      expect(actions.removeUserLastMessage()).toMatchSnapshot();
    });
  });

  describe('reducers handle actions', () => {
    const {
      actions: { addMessage, incMessage, removeUserLastMessage },
      reducer
    } = messagesHandler;
    let state = reducer(undefined, {});

    it('actions.addMessage', () => {
      expect(reducer(state, addMessage(item))).toMatchSnapshot();
    });
    it('actions.incMessage', () => {
      let state = reducer(undefined, {});
      expect(reducer(state, incMessage(item))).toMatchSnapshot();
    });
    it('actions.removeUserLastMessage', () => {
      let state = reducer(undefined, {});
      expect(reducer(state, addMessage(item))).toMatchSnapshot();
      expect(
        reducer(state, removeUserLastMessage(item.username))
      ).toMatchSnapshot();
    });
  });
});
