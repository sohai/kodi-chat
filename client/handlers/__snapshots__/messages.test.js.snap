// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(handler) messages actions crators corect objects actions.addMessage 1`] = `
Object {
  "type": "ADD_MESSAGE",
}
`;

exports[`(handler) messages actions crators corect objects actions.incMessage 1`] = `
Object {
  "type": "INC_MESSAGE",
}
`;

exports[`(handler) messages actions crators corect objects actions.removeUserLastMessage 1`] = `
Object {
  "type": "REMOVE_USER_LAST_MESSAGE",
}
`;

exports[`(handler) messages have right stucture 1`] = `
Object {
  "actions": Object {
    "addAndSendMessage": [Function],
    "addMessage": [Function],
    "incMessage": [Function],
    "removeMyLastMessage": [Function],
    "removeUserLastMessage": [Function],
  },
  "reducer": [Function],
}
`;

exports[`(handler) messages reducers handle actions actions.addMessage 1`] = `
Array [
  Object {
    "message": "test",
    "username": "username",
  },
]
`;

exports[`(handler) messages reducers handle actions actions.incMessage 1`] = `
Array [
  Object {
    "message": "test",
    "username": "username",
  },
]
`;

exports[`(handler) messages reducers handle actions actions.removeUserLastMessage 1`] = `
Array [
  Object {
    "message": "test",
    "username": "username",
  },
]
`;

exports[`(handler) messages reducers handle actions actions.removeUserLastMessage 2`] = `Array []`;
